src/coprocessor_v2/plugin_registry.rs

```diff
@@ -307,18 +302,18 @@ impl PluginRegistryInner {
     pub fn load_plugin<P: AsRef<OsStr>>(
         &mut self,
         filename: P,
-    ) -> Result<&'static str, PluginLoadingError> {
+    ) -> Result<String, PluginLoadingError> {
         let plugin = unsafe { LoadedPlugin::new(&filename) };
         if let Err(err) = &plugin {
             let filename = filename.as_ref().to_string_lossy();
             warn!("failed to load coprocessor plugin. Maybe not compiled correctly as a TiKV plugin?"; "plugin_path" => ?filename, "error" => ?err);
         }
         let plugin = plugin?;
 
-        let plugin_name = plugin.name();
+        let plugin_name = plugin.name().to_string();
 
         self.loaded_plugins.insert(
-            plugin_name.to_string(),
+            plugin_name.clone(),
```

Comment:
Does it work?

```suggestion
            plugin.name().to_string(),
```

Reply:
I think it should work. I moved the `to_string()` method call up a few lines (https://github.com/tikv/tikv/pull/10090/files#diff-e4204ab49a0a810cb8c0e685991d164e985fb6207a4106161a7aaf678321cc79R313), so we are cloning a `String` here.